tests/Unit/Pypp/PyppFundamentals.hpp

```diff
@@ -563,6 +583,45 @@ struct ToPyObject<T, Requires<tt::is_a_v<Tensor, T> and
   }
 };
 
+template <typename T>
+struct ToPyObject<
+    T, Requires<tt::is_a_v<Tensor, T> and
+                cpp17::is_same_v<typename T::type, std::complex<double>>>> {
+  static PyObject* convert(const T& t) {
+    std::array<long, T::rank()> dims =
+        convert_array_of_size_t_to_array_of_long(T::index_dims());
+    // clang-tidy: cstyle casts, pointer arithmetic, implicit decay array to
+    // pointer. (Expanded from macro.)
+    PyObject* npy_array =
+        PyArray_SimpleNew(T::rank(), dims.data(), NPY_COMPLEX128);  // NOLINT
+    if (npy_array == nullptr) {
+      throw std::runtime_error{"Failed to create PyArray."};
+    }
+
+    for (IndexIterator<T::rank()> index_it((Index<T::rank()>(T::index_dims())));
+         index_it; ++index_it) {
+      const auto tensor_idx = (*index_it).indices();
+      *get_ptr_to_elem<typename T::type>(npy_array, tensor_idx) =
```

Comment:
Nitpick: T::type is required to be `std::complex<double>` so that can be explicitly used here.